### ISSUE TITLE
Document `SOLO.predict()` return values better.

### ISSUE THREAD

**Issue by alexlenail** [scverse/scvi-tools#2688]
The [documentation](https://docs.scvi-tools.org/en/stable/api/reference/scvi.external.SOLO.html#scvi.external.SOLO.predict
) reads:

> soft ([bool](https://docs.python.org/3/library/functions.html#bool) (default: True)) – Return probabilities instead of class label

However, the return values are not probabilities -- they range from -30 to 30 in my case. How should I interpret these? 

<img width="599" alt="image" src="https://github.com/scverse/scvi-tools/assets/2761597/aadcdd4f-4a2c-4c6c-b6e5-6ffd1f81dc2f">

**Comment by canergen** [comment]
It's indeed the logits on a per cell level. You can transform them with scipy.special.softmax(x, axis=1) to probabilities. @martinkim0 Can you fix the documentation?

**Comment by alexlenail** [comment]
`softmax` or `idxmax`?

https://github.com/scverse/scvi-tools/blob/95f2e1d2fa921c6433a04e257d844280ba0c25e5/scvi/external/solo/_model.py#L427-L430

**Comment by martinkim0** [comment]
@alexlenail Thanks for bringing this up! I'll take a look

**Comment by alexlenail** [comment]
I raised this issue because I got this strange result: the cells with the very smallest values of `doublet` and largest values of `singlet` appear to _also_ be doublets, by careful inspection. i.e. my cells appear to most likely be doublets if they are extreme values on either side of the distribution SOLO returns. This suggests the possibility of a deep issue in this implementation of SOLO. Is this the original implementation?

**Comment by martinkim0** [comment]
This is not the original implementation. You can find the original one here: https://github.com/calico/solo

**Comment by martinkim0** [comment]
> the cells with the very smallest values of doublet and largest values of singlet appear to also be doublets, by careful inspection. i.e. my cells appear to most likely be doublets if they are extreme values on either side of the distribution SOLO returns

I need to look into this more since I didn't implement `SOLO` in the codebase, but from my intuition this is the desired behavior. Since the `SOLO` classifier will output two logits (one for singlets and one for doublets), we expect cells predicted to be doublets to have large values for one logit (the doublet one) and small values for the other logit (the singlet one). 

Is this what you were referring to or another issue? Feel free to share any plots/data if possible, and I can look into this on my end.

**Comment by alexlenail** [comment]
hi @martinkim0 
Alas it's worse than that. The cells which have the highest values of `doublet` (e.g. 10) _and_ the lowest values of `doublet` (e.g. -30) appear to be the credible doublets.

**Comment by canergen** [comment]
Is the original SOLO doing better or is this a problem with SOLO? I myself found scrublet to perform better in some datasets (it's very much dataset dependent in my hands). You should see smooth values in cellular neighborhoods (similar cells should have values close to each other).
